File: VB/E5137/EndPointHelper.vb
Imports System
Imports System.ServiceModel
Imports DevExpress.Xpo.DB

Namespace E5137

    Public Class EndPointHelper

        Public Shared Function GetDataStore(ByVal connectionString As String) As IDataStore
            Dim address As EndpointAddress = New EndpointAddress(connectionString)
            Dim binding As BasicHttpBinding = New BasicHttpBinding()
            binding.MaxReceivedMessageSize = Integer.MaxValue
            binding.ReaderQuotas.MaxArrayLength = Integer.MaxValue
            binding.ReaderQuotas.MaxDepth = Integer.MaxValue
            binding.ReaderQuotas.MaxBytesPerRead = Integer.MaxValue
            binding.ReaderQuotas.MaxStringContentLength = Integer.MaxValue
            Try
                Dim store As IDataStore = New DataStoreClient(binding, address)
                store.AutoCreateOption.ToString()
                Return store
            Catch e As Exception
                Throw New Exceptions.UnableToOpenDatabaseException(connectionString, e)
            End Try
        End Function
    End Class
End Namespace

File: CS/E5137/EndPointHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.ServiceModel;
using DevExpress.Xpo.DB;


namespace E5137 {
    public class EndPointHelper {
        public static IDataStore GetDataStore(string connectionString) {
            EndpointAddress address = new EndpointAddress(connectionString);
            BasicHttpBinding binding = new BasicHttpBinding();
            binding.MaxReceivedMessageSize = Int32.MaxValue;
            binding.ReaderQuotas.MaxArrayLength = Int32.MaxValue;
            binding.ReaderQuotas.MaxDepth = Int32.MaxValue;
            binding.ReaderQuotas.MaxBytesPerRead = Int32.MaxValue;
            binding.ReaderQuotas.MaxStringContentLength = Int32.MaxValue;
            try {
                IDataStore store = new DataStoreClient(binding, address);
                store.AutoCreateOption.ToString();
                return store;
            }
            catch(Exception e) {
                throw new DevExpress.Xpo.DB.Exceptions.UnableToOpenDatabaseException(connectionString, e);
            }
        }
    }
}
